contentcuration/contentcuration/views/base.py

```diff
@@ -61,6 +64,26 @@
 CURRENT_USER = "current_user"
 
 
+def _get_public_channel_languages():
+    public_channel_query = Language.objects.filter(channel_language__public=True,
```

Comment:
We want to get all the languages that share a lang code of a language of the public channels - I think it's probably simpler for this query just to return these languages, and then on the frontend, we filter languages by the lang_codes returned from here.

I would suggest just returning just the ids from this initial query to the frontend.

Reply:
Have made a suggestion above about how to handle this in the javascript - we could also cache this language list and blow it away at public channel publish like we do public channel caching.